### PR TITLE
Add oxford comma to list of contributors

### LINKED ISSUE #57: Create addOxfordComma
## Background

The `pull-requests` controller has a method called `updateContributorsList` ([see code](https://github.com/ember-learn/whats-new-in-emberland/blob/0d4b187a630abfd21fbd1bc31d976596187aed69/app/controllers/pull-requests.js#L80-L97)).

Given an array of `user`'s (an array of POJOs), the method creates a string and saves it to `contributorsList`, a tracked property. The Ember Times editors can copy-paste that string to a Markdown file, so that a list of links gets rendered in an Ember Times issue:

An Ember Times issue shows a list of contributors

## Problem

In the screenshot, notice the word `and` that appears between `@toh995` and `Yehuda Katz (@wycats)`. Currently, the editors have to manually add the word.

## TODOs

- [ ] In the controller, create a method called `addOxfordComma`. As input, it takes an array of strings. As output, it returns a string with an Oxford comma.

    In pseudocode,

    ```
    addOxfordComma(words = [])

    Case One: 0, 1 or 2 words
    Returns '', 'X', or 'X and Y'

    Case Two: 3 or more words
    Returns '..., X, Y, and Z'
    ```

- [ ] Add unit tests to check that `addOxfordComma` works.

- [ ] Update the method `updateContributorsList` to call `addOxfordComma`. Check that the conjunction `and` appears when you run the app locally.

- [ ] Update the existing unit test for `updateContributorsList` so that the test passes again.

### PR DESCRIPTION
**Pull request by JamesS-M**
I think I wrote the the tests in the correct places 😛 
Closes #57

### CODE REVIEW

File: app/controllers/pull-requests.js [hunk 1]
```diff
@@ -91,8 +91,14 @@ export default class PullRequestsController extends Controller {
...
 
-    this.contributorsList = contributorsList;
+    this.contributorsList = this.addOxfordComma(contributorsList)
+  }
+
+  addOxfordComma(contributors = []) {
+    if (contributors.length === 2) {
+      return contributors.join(' and ');
+    }
+    return [contributors.slice(0, -1).join(', '), contributors.pop()].filter(contributor => contributor !== '').join(', and ');
```
Comment (ijlee2) [inline app/controllers/pull-requests.js:102]: I liked your use of `slice()` to get the first `n - 1` words.

Can we update the code to be general (use the name `words`) and non-mutating (avoid `pop()`)? In this case, we may assume that each word is a non-empty string and skip calling `filter()`.

```suggestion
  addOxfordComma(words = []) {
    if (words.length <= 2) {
      return words.join(' and ');
    }

    return [...words.slice(0, -1), `and ${words[words.length - 1]}`].join(', ');
```

File: app/controllers/pull-requests.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ export default class PullRequestsController extends Controller {
   }
 
   updateContributorsList(users) {
-    const contributorsList = users
+    let contributorsList = users
```
Comment (ijlee2) [inline app/controllers/pull-requests.js:81]: Can we revert this change? (I'm guessing you might have used `let` to indicate that the `pop()` method will mutate `contributorsList`.)

```suggestion
    const contributorsList = users
```

File: tests/unit/controllers/pull-requests-test.js [hunk 3]
```diff
@@ -158,12 +158,61 @@ module('Unit | Controller | pull-requests', function(hooks) {
...
   });
+
+  test('addOxfordComma works', function (assert) {
+    const controller = this.owner.lookup('controller:pull-requests');
+
+    controller.updateContributorsList([]);
+    assert.strictEqual(
+      controller.contributorsList,
+        ''
+      ,
```
Comment (ijlee2) [inline tests/unit/controllers/pull-requests-test.js:174]: Minor.

```suggestion
      '',
```

File: tests/unit/controllers/pull-requests-test.js [hunk 4]
```diff
@@ -158,12 +158,61 @@ module('Unit | Controller | pull-requests', function(hooks) {
...
+      'We return an empty string for no contributors.'
+    );
+
```
Comment (ijlee2) [inline tests/unit/controllers/pull-requests-test.js:177]: Can we add 1 more assertion to test the case of 1 contributor?

**Review by ijlee2 — DISMISSED** [review]
@JamesS-M Thank you for working on this issue quickly and making your 1st contribution to `whats-new-in-emberland`! 🎉

The code changes look great overall. I made a few suggestions that I think will help us maintain the app better. Before I merge your pull request, will you go over the suggestions and push a commit that addresses them?

If you have questions or want clarification about my feedback, please feel free to comment. Thanks again!

File: tests/unit/controllers/pull-requests-test.js (hunk 4 above)
Comment (JamesS-M) [reply tests/unit/controllers/pull-requests-test.js:176]: Yes good call.

**Comment by JamesS-M** [comment]
> @JamesS-M Thank you for working on this issue quickly and making your 1st contribution to `whats-new-in-emberland`! 🎉
> 
> The code changes look great overall. I made a few suggestions that I think will help us maintain the app better. Before I merge your pull request, will you go over the suggestions and push a commit that addresses them?
> 
> If you have questions or want clarification about my feedback, please feel free to comment. Thanks again!

@ijlee2 I appreciate your detailed feedback!

**Review by ijlee2 — APPROVED** [review]
Thanks again!
